wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,152 @@ export default {
           }
         }
       })(_this);
-    }
+    },
+
+    process: function(data, query, headers) {
+      // If there are posts for this query, map them to the template.
+      let posts = (Array.isArray(data)) ?
+        data.map(this.maps()[this.type]) : false;
+
+      // Set posts and store a copy of the query for reference.
+      this.$set(this.posts[query.page], 'posts', posts);
+      this.$set(this.posts[query.page], 'headers', Object.freeze(headers));
+
+      // If there are no posts, pass along to the error handler.
+      if (!Array.isArray(data))
+        this.error({error: data, query: query});
+
+      this.$set(this, 'init', true);
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+        this.totalPages = [2,3,4,5]
+      } 
+    },
+
+    updatePagination: function(){
+      this.scrollToTop();
+      let url = [
+        this.domain,
+        this.lang.path,
+        this.endpoints[this.type],
+        this.buildUrlQuery(this.query)
+      ].join('');
+
+      fetch(url)
+      .then(this.response)
+      .then(data => {
+      let query = this.query;
```

Comment:
Minor: can you indent lines 271 - 290 so it's clearer that they're inside the `then(data => { })` block?